api/v1beta1/openstackdataplanedeployment_types.go

```diff
@@ -100,7 +100,7 @@ func (instance *OpenStackDataPlaneDeployment) InitConditions() {
 
 	if instance.Spec.NodeSets != nil {
 		for _, nodeSet := range instance.Spec.NodeSets {
-			readyCondition := condition.Type(fmt.Sprintf(NodeSetDeploymentReadyCondition, nodeSet))
+			readyCondition := condition.Type(fmt.Sprintf(NodeSetDeploymentReadyMessage, nodeSet))
```

Comment:
this is what I'm referring to ^^. You changed NodeSetDeploymentReadyCondition to be condition.Type, but then you just switch back to using the string here.

Reply:
Right, I see what you mean. I'll have a re-think about this entire PR. I might close it and start a fresh one with a more wholistic approach specifically around the conditions. Rather than trying to tack it on with my functional tests